k2/python/k2/fsa_algo.py

```diff
@@ -50,37 +51,55 @@ def top_sort(fsa: Fsa) -> Fsa:
     for name, value in fsa.named_tensor_attr():
         setattr(sorted_fsa, name, value.index_select(0, arc_map))
     for name, value in fsa.named_non_tensor_attr():
-        setattr(sorted_fsa, name, deepcopy(value))
+        setattr(sorted_fsa, name, value)
     return sorted_fsa
 
 
 def intersect(a_fsa: Fsa, b_fsa: Fsa) -> Fsa:
     '''Compute the intersection of two FSAs.
 
-    Note:
-      Intersection is also known as compose in FST.
-
     Args:
       a_fsa:
         The first input FSA. It can be either a single FSA or a vector of FSAs.
       b_fsa:
         The second input FSA. It can be either a single FSA or a vector of FSAs.
 
+    CAUTION:
+      ``aux_labels`` are discarded since it is not clear how to set the
```

Comment:
We don't need to treat aux_labels specially here.  The rules will be:
    For attributes where only one source (a or b) has that attribute:
       Copy via arc_map, or use zero if arc_map has -1.  This rule works for both 
       floating point and integer.
    For attributes where both sources (a and b) have that attribute:
       For floating point attributes: sum via arc_maps, or use zero if arc_map has -1.  
       For integer attributes, it's not supported for now.
Later we can add an optional argument to `intersect` that is a dict from an attribute-name to some
kind of object or enum or string that says how to process that attribute.

Reply:
Will fix it.

Reply:
Fixed.